### PR TITLE
[Fix]: scan-app-data conversion

### LINKED ISSUE #34283: [Bug]: scan-app-data conversion
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

After running occ files:scan-app-data on server v24.0.5, I'm getting 

In ScanAppData.php line 242:
Implicit conversion from float 0.4666666666666667 to int loses precision 

### Steps to reproduce

1. Install nextcloud
2. run occ files:scan-app-data
3. get error message

### Expected behavior

Not getting error message 

### Installation method

Community Manual installation with Archive

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Nginx

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated to a major version (ex. 22.2.3 to 23.0.1)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

```shell
Enabled:
  - accessibility: 1.10.0
  - activity: 2.16.0
  - admin_audit: 1.14.0
  - bruteforcesettings: 2.4.0
  - cloud_federation_api: 1.7.0
  - comments: 1.14.0
  - dav: 1.22.0
  - federatedfilesharing: 1.14.0
  - files: 1.19.0
  - files_downloadactivity: 1.13.0
  - files_sharing: 1.16.2
  - files_trashbin: 1.14.0
  - lookup_server_connector: 1.12.0
  - provisioning_api: 1.14.0
  - serverinfo: 1.14.0
  - settings: 1.6.0
  - theming: 1.15.0
  - viewer: 1.8.0
  - workflowengine: 2.6.0
```

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by EWouters**
Adapted from sugestions by @Hiyoal in #34283 and @st3iny in #35935.

See https://github.com/nextcloud/server/issues/34283#issuecomment-1288075064
and https://github.com/nextcloud/server/pull/35935#discussion_r1059905594.

Closes #34283.

Co-authored-by: Richard Steinmetz <richard@steinmetz.cloud>
Signed-off-by: EWouters <6179932+EWouters@users.noreply.github.com>

* Resolves: #34283

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files/lib/Command/ScanAppData.php [hunk 1]
```diff
@@ -239,7 +240,7 @@ protected function showSummary($headers, $rows, OutputInterface $output) {
 	protected function formatExecTime() {
 		$secs = round($this->execTime);
 		# convert seconds into HH:MM:SS form
-		return sprintf('%02d:%02d:%02d', ($secs / 3600), ($secs / 60 % 60), $secs % 60);
+		return sprintf('%02d:%02d:%02d', intdiv($secs, 3600), intdiv($secs, 60) % 60, $secs % 60);
```
Comment (st3iny) [inline apps/files/lib/Command/ScanAppData.php:243]:
```suggestion
		return sprintf('%02d:%02d:%02d', (int)($secs / 3600), (int)($secs / 60) % 60, $secs % 60);
```

Just converting to int is fine here too. (Similar to https://github.com/nextcloud/server/pull/32059)

This should also fix static analysis.

Comment (Hiyoal) [reply apps/files/lib/Command/ScanAppData.php:243]: I may be wrong, but this was the first thing I tried which still threw the error:
"Implicit conversion from float 0.4666666666666667 to int loses precision"

The only thing that worked was after I used intdiv. Seems odd though as I would assume statically using int should have fixed the issue, but then again I'm just a nuffy who is a beginner hobbyist coder at best!

Comment (st3iny) [reply apps/files/lib/Command/ScanAppData.php:243]: I just tested this and you are right. The problematic part is the modulo operator as it implicitly converts to int.

I edited my suggestion and now it should prevent any deprecation warning.

**Comment by come-nc** [comment]
Superseeded by https://github.com/nextcloud/server/pull/32059

**Comment by st3iny** [comment]
> Superseeded by #32059

No. This is about apps/files/lib/Command/ScanAppData.php. You linked the PR that fixed apps/files/lib/Command/Scan.php.

**Comment by come-nc** [comment]
Sorry about that

**Comment by EWouters** [comment]
Should I squash and force push or it it fine like this?

**Comment by st3iny** [comment]
> Should I squash and force push or it it fine like this?
> 
> (I used only the browser and my phone for these changes 🤓 (and I tested them locally))

Please squash your commits.

**Comment by st3iny** [comment]
> 

I restarted failed tests. This might be a transient failure.

**Review by st3iny — APPROVED** [review]
LGTM! Cypress failures are unrelated.

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**